common/utils/src/main/java/org/apache/spark/SparkThrowable.java

```diff
@@ -35,19 +35,29 @@
  */
 @Evolving
 public interface SparkThrowable {
-  // Succinct, human-readable, unique, and consistent representation of the error category
-  // If null, error class is not set
-  String getErrorClass();
+  /**
+   * Succinct, human-readable, unique, and consistent representation of the error condition.
+   * If null, error condition is not set.
+   */
+  String getCondition();
```

Comment:
shall we say `getErrorCondition`?

Reply:
IMHO, the `error` word is unnecessary. The interface is about errors, and I don't think we should add `error` everywhere. In this logic we should add the error to all methods:
- `getErrorMessageParameters`
- `getErrorQueryContext`
- `getErrorSqlContext`

And when you call `exception.getCondition`, do you expect a condition of somewhat else of the exception/error?

Reply:
@srielau WDYT?

Reply:
BTW, the SQL standard just uses the word `condition`:
<img width="920" alt="Screenshot 2024-10-09 at 19 38 51" src="https://github.com/user-attachments/assets/bd17ce98-3f99-4419-a3aa-927cf162d696">